backend/editor/OT/docs/operation_parsing.md

```diff
@@ -0,0 +1,106 @@
+# Operation Parsing & Application
+
+Surprisingly most of the complexity in the OT editor lives in the operation parsing and application logic. This document aims to help demystify some of it. As a quick aside, everything regarding operations can be found in the `operation` folder.
+
+## The operation struct
+Operations (in Go) are defined as a struct. The operations we receive from the client conform to this structure and are parsed using a JSON parsing library (more on that later as there is a bit of complexity behind this). 
+```go
+type (
+	// OperationModel defines an simple interface an operation must implement
+	OperationModel interface {
+		TransformAgainst(op OperationModel, applicationType EditType) (OperationModel, OperationModel)
+		Apply(parentNode cmsjson.AstNode, applicationIndex int, applicationType EditType) (cmsjson.AstNode, error)
+	}
+
+	// Operation is the fundamental incoming type from the frontend
+	Operation struct {
+		Path                  []int
+		OperationType         EditType
+		AcknowledgedServerOps int
+
+		IsNoOp    bool
+		Operation OperationModel
+	}
+)
+// EditType is an enum with `int` as the base type
+type EditType int
+const (
+	Add EditType = iota
+	Remove
+)
+```
+The above code snippet uniquely defines an operation. Operations take a path to where in the document AST they are being applied (see the paper on tree based transform functions) and the physical operation being applied. The operation being applied (`OperationModel`) is actually an interface and is the reason why parsing operations is more complex than it seems. This interface defines two functions, one for transforming a operation against another operation and one for applying an operation to an AST. 
```

Comment:
would be nice to link the papers in the docs themselves, saves anyone reading having to click back and forth